src/Serializer/AbstractCollectionNormalizer.php

```diff
@@ -54,11 +57,28 @@ public function supportsNormalization(mixed $data, string $format = null, array
         return static::FORMAT === $format && is_iterable($data);
     }
 
+    /**
+     * @deprecated since Symfony 6.3, use "getSupportedTypes(?string $format)" instead
+     */
     public function hasCacheableSupportsMethod(): bool
     {
+        trigger_deprecation('symfony/serializer', '6.3', 'The "%s()" method is deprecated, use "getSupportedTypes()" instead.', __METHOD__);
+
         return true;
     }
 
+    public function getSupportedTypes(?string $format)
+    {
+        return match($format) {
+            static::FORMAT => [
```

Comment:
I'm really not sure about this modification.

I've tried to understand the purpose of the getSupportedTypes method and adapted it to our class.

I'm waiting for your feedback to continue on the subject.

Reply:
See https://github.com/api-platform/core/pull/5611#discussion_r1221524497